beancount.el

```diff
@@ -753,15 +753,20 @@ what that column is and returns it (an integer)."
 
 (defun beancount--account-currency (account)
   ;; Build a regexp that matches an open directive that specifies a
-  ;; single account currencydaaee. The currency is match group 1.
-  (let ((re (concat "^" beancount-date-regexp " +open"
-                    "\\s-+" (regexp-quote account)
-                    "\\s-+\\(" beancount-currency-regexp "\\)\\s-+")))
-    (save-excursion
-      (goto-char (point-min))
-      (when (re-search-forward re nil t)
-        ;; The account has declared a single currency, so we can fill it in.
-        (match-string-no-properties 1)))))
+  ;; single account currency. The currency is match group 1.
+  (save-excursion
+    (goto-char (point-min))
+    (and (or (re-search-forward
+              (concat "^" beancount-date-regexp " +open"
+                      "\\s-+" (regexp-quote account)
+                      "\\s-+\\(" beancount-currency-regexp "\\)\\>")
+              nil t)
+             (re-search-forward
+              (concat "^option\\s-+\"operating_currency\"\\s-+\"\\("
+                      beancount-currency-regexp
+                      "\\)\"")
+              nil t))
+         (match-string-no-properties 1))))
```

Comment:
If I read it right, this would use the operating currency for accounts that specify more than one possible currency, regardless whether it is or not in the list of allowed currencies. I am not sure this would be the desired behavior.

Reply:
No, it would use operating_currency as a fallback, for accounts that specify _no_ currencies at all. It will use the first specified currency otherwise (which is the original behavior).

Reply:
Oh, I see now that you changed the regexp. This is a change in behavior: before the currency was automatically picked up only when only one currency was specified (and it still says so in the comment). I don't think one behavior is better than the other, but changing it will break someone else use case. Having to choose, I think it is preferable to preserve the existing behavior.